contents/regex/Regex.md

```diff
@@ -0,0 +1,368 @@
+> * 02-25-2017 Workshop
+
+However, you want them to be written in (DD-MM-YYYY) instead. As we can see, this isn't as simple as just finding and replacing text, we must also maintain the data in the search string (e.g. `02-20` becomes `20-02`). This can be done using a **capturing group**. A capturing group is a part of a Regex surrounded by brackets `()`. They are numbered `1`, `2`, `3`, from left to right and are used to hold captured data. For example:
+
+`(\d\d)-(\d\d)-(\d\d\d\d)` against `02/20/2017` will result in group 1 : `02`, group 2 : `20`, and group 3 : `2017`.
+
+Using this, we can construct our regex as follows:
+
+* Find: `(\d\d)-(\d\d)-(\d\d\d\d)`
+* Replace: `\2-\1-\3`
+
+In the replace field, `\1` represents the group matched in the string. For example, `\1` corresponds to `02`, `\2` corresponds to `20` and so on. As you can see, we are swapping group 1 (the month) and group 2 (the day) to get the desired result.
+
+### Result
+
+> Calendar <br>
+> * 20-02-2017 Paul's Birthday <br>
+> * 24-02-2017 Presentation at work <br>
+> * 25-02-2017 Workshop
+
+## Further exploration
+
+These examples provide a quick look as to what Regex can be used for, without code. You can also use Regex as part of many different programming languages, to search strings within the code. Many languages implement it differently, for example Python uses the [`re`](https://docs.python.org/2/library/re.html) library, Java uses [`Pattern`](http://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html) and Javascript has regex built into the language using [`/` notation](https://developer.mozilla.org/en/docs/Web/JavaScript/Guide/Regular_Expressions). You can search the respective language documentation for a more comprehensive view of how to use Regex in the language of your choice.
+
+For more examples of what Regex can do, you can refer to the tutorial links in the appendix.
+
+## Quick Reference
+
+Name | Regex | Use | Example
+--- | --- | --- | ---
+Literal characters | Any character(s) | Matches any substring | `cat` matches `category`
+Escape characters | `\t`, `\n`, `\d`... | Matches [special characters](http://www.regular-expressions.info/refcharacters.html). | `\t` matches a tab character
+Anchors | `^`, `$` | Matches start and end of string | `^ant` matches `antics` but not `pant`
+Charcter classes | `[abc]` | Matches any character in the character class | `l[ai]st` matches `last` and `list`
+Quantifiers | `+` | Matches one or more of the previous character | `no+` matches `nooooooo` but not `n`
+ | `?` | Matches zero or one of the previous character | `colou?r` matches `color` and `colour`
+ | `*` | Matches zero or more of the previous character | `ba*` matches `b`, `ba` and `baaa`
+ | `{1, 3}` or `{2}` | Matches the previous character a specified number of times | `a{1, 3}` matches `a`, `aa` and `aaa`, `a{2}` matches `aa`
+Groups | `(...)` | Groups a pattern of data | `Used for extracting data, see below`
```

Comment:
"Used for extracting data, see below" needs not be surrounded by backtick.
Also, what is "below"? I think you're trying to say "above" where you demonstrated the usage of capturing group.